app/src/main/java/aloksharma/ufl/edu/stash/RegisterActivity.java

```diff
@@ -0,0 +1,89 @@
+package aloksharma.ufl.edu.stash;
+
+import android.app.ProgressDialog;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.ActionBarActivity;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import com.parse.ParseException;
+import com.parse.ParseUser;
+import com.parse.SignUpCallback;
+
+public class RegisterActivity extends ActionBarActivity implements View.OnClickListener{
+
+    Button registerButton;
+    public EditText username,password,retypepassword;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_register);
+
+        username = (EditText) findViewById(R.id.username);
+        password = (EditText) findViewById(R.id.password);
+        retypepassword = (EditText) findViewById(R.id.retypepassword);
+
+        registerButton = (Button) findViewById(R.id.registerButton);
+        registerButton.setOnClickListener(this);
+    }
+
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()){
+            case R.id.registerButton:
+                boolean validationError = false;
+                StringBuilder errorMessage = new StringBuilder("Please ");
+
+                if(username.getText().toString().equals("") || password.getText().toString().equals("") || retypepassword.getText().toString().equals("")){
+                    validationError = true;
+                    errorMessage = errorMessage.append("fill all the text fields.");
+                }
+
```

Comment:
Try to avoid stray multiple blank lines. There arent too many till now, so its fine for now.